static-site/components/list-resources/listResourcesApi.ts

```diff
@@ -0,0 +1,335 @@
+import { ResourceType, Resource, Group, PathVersionsForGroup, FetchGroupHistory } from "./types";
+import { InternalError } from "../error-boundary";
+
+interface ResourceListingDatasets {
+  pathPrefix: string;
+  pathVersions: Record<string, string[]>;
+}
+
+interface ResourceListingIntermediates {
+  pathPrefix: string;
+  latestVersions: {
+    [id: string]: {
+      [filename: string]: [url: string, mostRecentlyIndexed: string]
+    }
+  }
+}
+
+/**
+ * A callback function used by the <ListResources> component to fetch information
+ * about the resources to list from the host server's /list-resources API endpoint.
+ */
+export async function listResourcesAPI(
+  sourceId: string,
+  resourceType: ResourceType,
+  {versioned, groupDisplayNames, groupUrl}: {
+    /** Report prior versions of each resource.
+     * TODO: infer this from the API data itself
+     */
+    versioned: boolean,
+    groupDisplayNames?: Record<string, string>,
+    groupUrl?: (groupName: string) => string
+  }
+): Promise<Group[]> {
+  const requestPath = `/list-resources/${sourceId}/${resourceType}`;
+  const response = await fetch(requestPath, {
```

Comment:
why not use the `fetchAndParseJson` method here? It'll handle all the error checking and you can inject the type param and avoid the cast on L44.